gen3-integration-tests/services/fence.py

```diff
@@ -56,60 +79,229 @@ def get_url_for_data_upload(self, file_name: str, user: str) -> dict:
             "Content-Type": "application/json",
         }
         response = requests.post(
-            url=self.DATA_UPLOAD_ENDPOINT,
+            url=f"{self.BASE_URL}{self.DATA_UPLOAD_ENDPOINT}",
             data=json.dumps({"file_name": file_name}),
             auth=auth,
             headers=headers,
         )
         return response
 
-    def has_url(self, response: dict) -> None:
-        assert "url" in response.keys(), f"URL key is missing.\n{response}"
-
-    def has_no_url(self, response: bytes) -> None:
-        assert (
-            "url" not in response.content.decode()
-        ), f"URL key is missing.\n{response}"
+    def get_url_for_data_upload_for_existing_file(self, guid: str, user: str) -> dict:
```

Comment:
The addition of docstrings would get this module more consistent with the other modules under `services`.

Reply:
Sure let me add the docstrings for this function

Reply:
Looks good!